src/plugins/context-menu/videojs-contextmenu.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * @module plugin
+ */
+import videojs from 'video.js';
+
+const defaults = {
+  cancel: true,
+  sensitivity: 10,
+  wait: 500,
+  disabled: false
+};
+
+const EVENT_NAME = 'vjs-contextmenu';
+
+/**
+ * Abstracts a DOM standard event into a vjs-contextmenu event.
+ *
+ * @private
+ * @param  {Player} player
+ * @param  {Event} event
+ *         A triggering, native event.
+ * @return {Player}
+ */
+function sendAbstractedEvent(player, event) {
+  if (player.contextmenu.options.disabled) {
+    // videojs-contextmenu is disabled
+    return player;
+  }
+  const abstracted = {
+    target: player,
+    type: EVENT_NAME
+  };
+
+  [
+    'clientX',
+    'clientY',
+    'pageX',
+    'pageY',
+    'screenX',
+    'screenY'
+  ].forEach(k => {
+    abstracted[k] = event[k];
+  });
+
+  return player.trigger(abstracted);
+}
+
+/**
+ * Handles both touchcancel and touchend events.
+ *
+ * @private
+ * @param  {Event} e
+ */
+function handleTouchEnd(e) {
+  const current = this.contextmenu.current;
+
+  if (!current) {
+    return;
+  }
+
+  const wait = this.contextmenu.options.wait;
+
+  if (e.type === 'touchend' && Number(new Date()) - current.time >= wait) {
+    sendAbstractedEvent(this, e);
+  }
+
+  this.contextmenu.current = null;
+}
+
+/**
+ * Handles touchmove events.
+ *
+ * @private
+ * @param  {Event} e
+ */
+function handleTouchMove(e) {
+  const current = this.contextmenu.current;
+
+  if (!current) {
+    return;
+  }
+
+  const touch = e.touches[0];
+  const sensitivity = this.contextmenu.options.sensitivity;
+
+  // Cancel the current touch if the pointer has moved in either direction
+  // more than the sensitivity number of pixels.
+  if (
+    touch.screenX - current.screenX > sensitivity ||
+    touch.screenY - current.screenY > sensitivity
+  ) {
+    this.contextmenu.current = null;
+  }
+}
+
+/**
+ * Handles touchstart events.
+ *
+ * @private
+ * @param  {Event} e
+ */
+function handleTouchStart(e) {
+
+  // We only care about the first touch point.
+  if (this.contextmenu.current) {
+    return;
+  }
+
+  const touch = e.touches[0];
+
+  this.contextmenu.current = {
+    screenX: touch.screenX,
+    screenY: touch.screenY,
+    time: Number(new Date())
```

Comment:
What is that? look like a hack ... please replace it with 'new Date().getTime()'